xarray/core/dataset.py

```diff
@@ -438,6 +438,16 @@ def _subset_with_all_valid_coords(self, variables, coord_names, attrs):
 
         return self._construct_direct(variables, coord_names, dims, attrs)
 
+    @property
+    def _level_coords(self):
+        level_coords = OrderedDict()
+        for name in self._coord_names:
+            var = self.variables[name]
+            if name != var.dims[0]:
+                continue
+            level_coords.update(var.to_coord().get_level_coords())
+        return level_coords
```

Comment:
Am I missing something here? Wouldn't this also work without the `continue` statement?

```Python
for name in self._coord_names:
    var = self.variables[name]
    if name == var.dims[0]:
        level_coords.update(var.to_coord().get_level_coords())
    return level_coords
```